scripts/im_plot.sl

```diff
@@ -31,51 +30,61 @@ echo ___im plot___
 
 python $gmsim/workflow/scripts/cybershake/add_to_mgmt_queue.py $MGMT_DB_LOC/mgmt_db_queue $SRF_NAME IM_plot running $SLURM_JOB_ID
 
-res=`python $gmsim/visualization/im/spatialise_im.py $CSV_PATH $STATION_FILE_PATH --out_dir $OUTPUT_XYZ_DIR`
-exit_val=$?
-
-
-if [[ $exit_val == 0 ]]; then
-    echo "finished making csvs for im_plot"
-
-    echo "start plotting stations"
-    
-    failed=0
-    fail_msgs=()
-    success_msgs=()
-    for f in $OUTPUT_XYZ_DIR/*.xyz; do
-        cd $OUTPUT_XYZ_DIR
-        out_dir=$(basename $f | cut -d'.' -f1)
-        mkdir -p $out_dir
-        cd $out_dir
-        if [[ "$f" == *"real"* ]]; then
-          res2=$(python $gmsim/visualization/sources/plot_items.py -t $SRF_NAME --xyz $f -c "$SRF_PATH" --xyz-cpt hot --xyz-cpt-invert --xyz-transparency 30 --xyz-size 0.1 --xyz-cpt-labels `cat $OUTPUT_XYZ_DIR/im_order.txt` -n 2)
-        else
-          res2=$(python $gmsim/visualization/sources/plot_items.py -t $SRF_NAME --xyz $f -c "$SRF_PATH" --xyz-model-params $MODEL_PARAMS --xyz-grid --xyz-grid-search 12m --xyz-landmask --xyz-cpt hot --xyz-cpt-invert --xyz-grid-contours --xyz-transparency 30 --xyz-size 1k --xyz-cpt-labels `cat $OUTPUT_XYZ_DIR/im_order.txt` -n 2)
-        fi
-        exit_val2=$?
-        if [[ $exit_val2 != 0 ]]; then
-            failed=1
-            failed_msgs+=( "failed $res2" )
-        else
-            success_msgs+=( "successfully $res2" )
-        fi 
-    done
+COMPS = `cat $CSV_PATH |cut  -d , -f 2 |sort |uniq` # find what components are used
+COMPS=($(echo $COMPS |tr " " "\n")) # convert to an array of strings
+COMPS_LEN=${#COMPS[@]} # length of the array
+for (( i=1;i<$COMPS_LEN;i++ )); #${COMPS[0]} is the header "component"
+do 
+    COMP=${COMPS[$i]}
+    OUTPUT_XYZ_DIR=$3/$COMP
```

Comment:
I'd prefer to have all argparse at the start of the script - just so it's in one place?